src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/MatchingState.cs

```diff
@@ -95,21 +101,33 @@ internal List<(SymbolicRegexNode<TSet> Node, DerivativeEffect[] Effects)> NfaNex
             return Node.CreateNfaDerivativeWithEffects(builder, minterm, context);
         }
 
+        /// <summary>
+        /// TODO: This method is only used to speed up the existing architecture, ideally should be redesigned
+        /// Use <see cref="SymbolicRegexMatcher{TSet}.IsNullableWithContext"/>
+        /// whereever possible
+        /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool IsNullableFor(uint nextCharKind)
+        {
+            return ((1 << (int)nextCharKind) & NullabilityInfo) != 0;
+        }
+
+        /// <summary>
+        /// Full nullability check for initialization
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal bool IsNullableForInit(uint nextCharKind)
```

Comment:
It looks like this is only used by BuildNullabilityInfo. Could this just be inlined into it rather than having this method defined here?